### ISSUE TITLE
device DAX/librpmem: cannot create remote replica on dev DAX 

### ISSUE THREAD

**Issue by kkajrewicz** [pmem/issues#484]
pool set file on client:
```
$cat pool.set
PMEMPOOLSET
AUTO /dev/dax0.0
REPLICA 192.168.0.181 remote.set
```
pool set file on server:
```
$cat remote.set
PMEMPOOLSET
AUTO /dev/dax0.0
```
Create pool with remote replica:
```
$pmempool create obj pool.set
(...)
<librpmem>: <3> [rpmem.c:232 rpmem_common_init] out-of-band connection established
<librpmem>: <4> [rpmem_obc.c:495 rpmem_obc_create] sending create request message
<librpmem>: <3> [rpmem_obc.c:503 rpmem_obc_create] create request message sent
<librpmem>: <4> [rpmem_obc.c:504 rpmem_obc_create] receiving create request response
<librpmem>: <3> [rpmem_obc.c:513 rpmem_obc_create] create request response received
<librpmem>: <1> [rpmem_obc.c:252 rpmem_obc_check_hdr_resp] Fatal error
<librpmem>: <1> [rpmem.c:450 rpmem_create] create request failed: Remote I/O error
<libpmemobj>: <1> [set.c:1348 util_poolset_remote_open] creating remote replica #1 failed
<libpmemobj>: <3> [set.c:2110 util_replica_close] set 0x1887440 repidx 0
<libpmemobj>: <3> [set.c:427 util_unmap_part] part 0x1887390
<libpmemobj>: <4> [set.c:430 util_unmap_part] munmap: addr 0x7fe443a00000 size 1054867456
<libpmemobj>: <3> [set.c:2110 util_replica_close] set 0x1887440 repidx 1
<libpmemobj>: <4> [set.c:2118 util_replica_close] freeing volatile header of remote replica #1
<libpmemobj>: <3> [set.c:567 util_poolset_close] set 0x1887440 del 1
<libpmemobj>: <3> [set.c:2110 util_replica_close] set 0x1887440 repidx 0
<libpmemobj>: <3> [set.c:427 util_unmap_part] part 0x1887390
<libpmemobj>: <3> [set.c:2110 util_replica_close] set 0x1887440 repidx 1
<libpmemobj>: <4> [set.c:2118 util_replica_close] freeing volatile header of remote replica #1
<libpmemobj>: <3> [set.c:158 util_remote_unload]
<libpmemobj>: <3> [dlsym.h:83 util_dlclose] handle 0x1887540
<librpmem>: <3> [librpmem.c:85 librpmem_fini]
<libpmemobj>: <3> [set.c:465 util_poolset_free] set 0x1887440
<libpmemobj>: <2> [obj.c:1083 pmemobj_create] cannot create pool or pool set
error: '/root/pool.set2' -- creating remote replica #1 failed
error: creating pool file failed
<libpmempool>: <3> [libpmempool.c:81 libpmempool_fini]
(...)
```

Remote pool creation failed  unexpectedly and local device dax was left dirty(pool was created on it)
```
$pmempool create obj /dev/dax0.0
error: '/dev/dax0.0' -- Non-empty file detected
error: creating pool file failed

$pmempool info /dev/dax0.0
Part file:
path                     : /dev/dax0.0
type                     : device dax
size                     : 1054867456

POOL Header:
Signature                : PMEMOBJ [part file]
Major                    : 3
Mandatory features       : 0x0
Not mandatory features   : 0x0
Forced RO                : 0x0
Pool set UUID            : f5f3afc2-7911-4d8a-ad09-19a029d625e7
UUID                     : a6dec9f3-f09c-4bfc-94a3-1c778cac43e3
Previous part UUID       : a6dec9f3-f09c-4bfc-94a3-1c778cac43e3
Next part UUID           : a6dec9f3-f09c-4bfc-94a3-1c778cac43e3
Previous replica UUID    : 01df8216-534d-4d5b-8c5c-693fb629bb50
Next replica UUID        : 01df8216-534d-4d5b-8c5c-693fb629bb50
Creation Time            : Thu Mar 02 2017 12:34:05
Alignment Descriptor     : 0x000007f737777310[OK]
Class                    : ELF64
Data                     : 2's complement, little endian
Machine                  : AMD X86-64
Checksum                 : 0x9c5d35824f2ceed8 [OK]
```

It's worth to mention that it is possible to replicate local pool created on /dev/dax0.0 to regular file on remote host

Hosts setup:
Red Hat Enterprise Linux Server release 7.3 Beta (Maipo)
kernel: 4.8.10
libfabric: v1.4.0-76-g5a30b19
provider: verbs

Found in 1.2+wtp1-350-g6b58aeb

**Comment by janekmi** [comment]
It is the same issue as pmem/issues#473 and it is fixed by pmem/nvml#1698.

**Comment by kkajrewicz** [comment]
Verified on 1.2+wtp1-414-gd4d1313
